src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,74 @@
+                .flatMap(num -> Arrays.stream((num.split(","))))
+                .map(Integer::parseInt)
+                .filter(num -> num % 2 == 0)
+                .min(Integer::compareTo)
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list: "
+                        + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+
+        return IntStream.range(0, numbers.size())
+                .mapToObj(i -> i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
+                .filter(num -> num % 2 != 0)
+                .mapToDouble(num -> (double) num)
+                .average()
+                .orElseThrow(() -> new NoSuchElementException("Can't get average from list: "
+                        + numbers));
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(s -> s.getAge() > fromAge
+                        && s.getAge() <= toAge
+                        && s.getSex() == Person.Sex.MAN)
+                .toList();
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only people whose age is from `fromAge` and to `maleToAge` (for men)
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(s -> (s.getAge() >= fromAge
+                        && s.getAge() <= maleToAge
+                        && s.getSex() == Person.Sex.MAN)
+                        || (s.getAge() >= fromAge
+                        && s.getAge() <= femaleToAge
+                        && s.getSex() == Person.Sex.WOMAN))
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                .toList();
```
Be attentive to the comment, you fixed only one place